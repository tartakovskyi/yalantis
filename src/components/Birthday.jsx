import React  from 'react'


const Birthday = ({ user:{dob, firstName, lastName} }) => {

  return (
    <li>{firstName} {lastName}</li>
  )
}


export default Birthday