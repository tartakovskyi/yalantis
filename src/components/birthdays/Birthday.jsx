import React from "react";
import PropTypes from "prop-types";

const Birthday = ({ user: { dob, firstName, lastName }, month }) => {
  const DateString = () => {
    const d = new Date(dob);
    return ("0" + d.getDate()).slice(-2) + " " + month + " " + d.getFullYear();
  };

  return (
    <li>
      {firstName} {lastName} - <DateString />
    </li>
  );
};

Birthday.propTypes = {
  user: PropTypes.shape({
    dob: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  month: PropTypes.string.isRequired,
};

export default React.memo(Birthday);
