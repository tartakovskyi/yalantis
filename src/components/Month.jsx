import React from 'react'
import Birthday from './Birthday'


const Month = ({ name, users }) => {

  debugger

  return (
    <li className="month">
      <h4 className="month__name">{name}</h4>
      <ul>
        {users.map(user => <Birthday user={user} key={user.id} />)}
      </ul>
    </li>
  )
}


export default Month

