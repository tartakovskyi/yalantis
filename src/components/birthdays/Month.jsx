import React from "react";
import PropTypes from 'prop-types'
import Birthday from "./Birthday";

const Month = ({ name, users }) => {
  return (
    <li className="month">
      <h4 className="month__name">{name}</h4>
      <ul>
        {users.map((user) => (
          <Birthday user={user} month={name} key={user.id} />
        ))}
      </ul>
    </li>
  );
};

Month.propTypes = {
  name: PropTypes.string.isRequired,
  users: PropTypes.array.isRequired
};

export default Month;
