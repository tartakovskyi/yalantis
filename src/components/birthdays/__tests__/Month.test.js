import React from "react";
import { render } from "@testing-library/react";
import Birthday from "../Birthday";
import Month from "../Month";

jest.mock("../Birthday", () => () => <li data-testid="birthday">Birthday</li>);

const users = [
  {
    id: "5e00928df3f5ec0978d90ce7",
    firstName: "Madeleine",
    lastName: "Brooks",
    dob: "2019-05-12T01:20:34.085Z",
  },
  {
    id: "5e00928db89ff9c2559f9560",
    firstName: "Watson",
    lastName: "Good",
    dob: "2019-05-09T03:24:32.532Z",
  },
  {
    id: "5e00928def1a7e201d4b676e",
    firstName: "Benita",
    lastName: "Welch",
    dob: "2019-05-28T01:05:52.237Z",
  },
];

test("Month should render correctly", () => {
  const { container, getAllByTestId } = render(
    <Month name="May" users={users} />
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(getAllByTestId("birthday").length).toBe(Object.keys(users).length);
});
