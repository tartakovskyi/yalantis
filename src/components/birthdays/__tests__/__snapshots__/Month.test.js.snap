// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month should render correctly 1`] = `
<li
  class="month"
>
  <h4
    class="month__name"
  >
    May
  </h4>
  <ul>
    <li
      data-testid="birthday"
    >
      Birthday
    </li>
    <li
      data-testid="birthday"
    >
      Birthday
    </li>
    <li
      data-testid="birthday"
    >
      Birthday
    </li>
  </ul>
</li>
`;
